### PR TITLE
refactor: add new action 'release-cn-artifacts'

### PR DESCRIPTION
**Pull request by zyy17**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

1. Add new action 'release-cn-artifacts' to make the action of releasing artifacts to cn region idempotent(always can retry);

2. Refine naming: `release-artifacts` -> `publish-github-release`;

Please explain IN DETAIL what the changes are in this PR and why they are needed:

- Summarize your change (**mandatory**)
- How does this PR work? Need a brief introduction for the changed logic (optional)
- Describe clearly one logical change and avoid lazy messages (optional)
- Describe any limitations of the current code (optional)

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

**Comment by zyy17** [comment]
This PR can be reviewed only after the official release of v0.4.0 to avoid potential failure of the v0.4.0 release due to **insufficient testing**.

File: .github/actions/release-cn-artifacts/action.yaml [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+      env:
+        AWS_ACCESS_KEY_ID: ${{ inputs.aws-cn-access-key-id }}
+        AWS_SECRET_ACCESS_KEY: ${{ inputs.aws-cn-secret-access-key }}
+        AWS_DEFAULT_REGION: ${{ inputs.aws-cn-region }}
+      with:
+        max_attempts: ${{ inputs.upload-max-retry-times }}
+        timeout_minutes: ${{ inputs.upload-retry-timeout }}
+        # The bucket layout will be:
+        # releases/greptimedb
+        # ├── v0.1.0
```
Comment (daviderli614) [inline .github/actions/release-cn-artifacts/action.yaml:115]: suggestion: The bucket layout is releases/greptimedb/0.1.0/xxx, not have 'v' prefix

Comment (zyy17) [reply .github/actions/release-cn-artifacts/action.yaml:115]: @daviderli614 the `version` is allocated by `allocate-runners` job. I think adding `v` prefix will be consistent for all the artifacts(including binaries and images).

**Review by daviderli614 — APPROVED** [review]

**Review by killme2008 — APPROVED** [review]
LGTM 👍

**Review by sunng87 — APPROVED** [review]

**Pull request merged**